comms.go

```diff
@@ -209,12 +210,14 @@ func (d *Device) WriteRawImageToButton(btnIndex int, rawImg image.Image) error {
 
 func (d *Device) rawWriteToButton(btnIndex int, rawImage []byte) error {
 	// Based on set_key_image from https://github.com/abcminiuser/python-elgato-streamdeck/blob/master/src/StreamDeck/Devices/StreamDeckXL.py#L151
+
+	if Min(Max(btnIndex, 0), int(d.deviceType.numberOfButtons)) != btnIndex {
+		return errors.New(fmt.Sprintf("Invalid key index: %d", btnIndex))
+	}
+
 	pageNumber := 0
 	bytesRemaining := len(rawImage)
 
-	// Surely no image can be more than 20 packets...?
-	payloads := make([][]byte, 20)
```

Comment:
Setting this value was actually preventing the Mini from displaying the image or text I would place on the button. I'd be curious what this was needed for as the Python and Node APIs don't set anything like this.